src/librustc_trans/trans/mir/lvalue.rs

```diff
@@ -65,7 +66,28 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                     tcx.sess.bug(&format!("using operand temp {:?} as lvalue", lvalue)),
             },
             mir::Lvalue::Arg(index) => self.args[index as usize],
-            mir::Lvalue::Static(_def_id) => unimplemented!(),
+            mir::Lvalue::Static(def_id) => {
+                // There are two things that may happen here:
+                //  1) If the static item is defined in this crate, it will be translated, and we
+                //     return a pointer to the result.
+                //  2) If the static item is defined in another crate then we add (or reuse) a
+                //     declaration of an external global, and return a pointer to that.
+                let const_ty = self.mir.lvalue_ty(tcx, lvalue);
+                let llval = if let Some(node_id) = tcx.map.as_local_node_id(def_id) {
+                    // Case 1.
+
+                    // The LLVM global has the type of its initializer,
+                    // which may not be equal to the enum's type for
+                    // non-C-like enums.
```

Comment:
This isn't true; the type of the global returned by get_item_val is independent of the value of the initializer.  We fix things up later, in trans_crate.  See also #26630.